torch/csrc/jit/codegen/cuda/ir_base_nodes.h

```diff
@@ -95,6 +95,40 @@ struct TORCH_CUDA_API Statement {
   // Make sure this is an Expr and return it as an Expr*
   Expr* asExpr();
 
+  // notation idea for generic downcasts:
+  //  val->as<Int>()->value()
+  //
+  // instead of:
+  //  static_cast<Int*>(val)->value()
+  //  or
+  //  stm->asVal()
+  //
+  // PS. plus debug build checks!
+  //
+  template<class T>
```

Comment:
You mean throw in all build flavors? I'd very much like that, although it may be worth doing some sanity measurements to make sure the overhead is reasonable. Since I see a lot of static_cast uses this would be equivalent to switching to dynamic_cast (technically we can make it more efficient if we want strict type maches)

Here I just modeled this primarily as an alternative to static_cast, which is not safe (so as<T> brings in the additional benefit of debug checks). 

The key question is: do you have a preference for the notation (current vs. as<T>)?